src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -0,0 +1,716 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+#if UNIFIED
+using CoreFoundation;
+using Foundation;
+using Security;
+#else
+using MonoTouch.CoreFoundation;
+using MonoTouch.Foundation;
+using MonoTouch.Security;
+using System.Globalization;
+#endif
+
+#if SYSTEM_NET_HTTP
+namespace System.Net.Http 
+#else
+namespace Foundation
+#endif
+{
+	public class NSUrlSessionHandler : HttpClientHandler

```

Comment:
Our current handler subclass `HttpMessageHandler` because it's much more lightweight in size. The managed linker can do a lot better job that way and it makes using that handler smaller than the previous (managed or CFNetwork-based one).